docs/apis/calcite.md

```diff
@@ -0,0 +1,195 @@
+
+## SQL APIs for Apache Apex
+
+Listed below are a couple of examples using java APIs which can be used by SQL/Apex users to create a DAG .
+
+## Pure Style SQL Application:
+
+With Apache Calcite Integration, you can also use SQL queries across different data sources and provide UDFs as per your business logic. This example will use Kafka as input source while HDFS file as the output destination.
+Following application code will be used to explain APIs.
+
+     public class PureStyleSQLApplication implements StreamingApplication
+     {
+       @Override
+       public void populateDAG(DAG dag, Configuration conf)
+       {
+         SQLExecEnvironment sqlEnv = SQLExecEnvironment.getEnvironment();
+    
+         sqlEnv = sqlEnv.registerTable("ORDERS", new KafkaEndpoint("localhost:9090", 
+                                        "inputTopic", new CSVMessageFormat(<<csvSchema>>)));
+    
+         sqlEnv = sqlEnv.registerTable("SALES", new FileEndpoint("/tmp/output", 
+                                       "out.file", new CSVMessageFormat(<<csvSchema>>)));
+    
+         sqlEnv = sqlEnv.registerFunction("APEXCONCAT", PureStyleSQLApplication.class, 
+                                                                  "apex_concat_str");
+    
+         sqlEnv.executeSQL(dag, sql);
+       }// populateDAG finished
+       
+       public static String apex_concat_str(String s1, String s2)
+       {
+            return s1 + s2;
+        }
+     }
+      
+### Constructing SQLExecEnvironment
+We will go over APIs from above code one by one. Class _**SQLExecEnvironment**_ provides a starting point and simple way to define metadata needed for running SQL statement. SQLExecEnvironment.getEnvironment() creates instance of SQLExecEnvironment.  
+
+    // Creates SQLExecEnvironment instance by using static method getEnvironment
+    SQLExecEnvironment sqlEnv = SQLExecEnvironment.getEnvironment();
+
+### Registering tables with SQLExecEnvironment
+Next, we need to register tables which can be used in a query. For this purpose, we can use _registerTable_ API from SQLExecEnvironment.
+
+    // Register KafkaEnpoint as table with kafka topic as table and specify data format as CSV
+    sqlEnv = sqlEnv.registerTable("ORDERS", new KafkaEndpoint("localhost:9090", 
+                                            "inputTopic", new CSVMessageFormat(<<csvSchema>>)));
+    
+    // Register FileEndpoint as tbale with HDFS file as table and specify data format as csv
+    sqlEnv = sqlEnv.registerTable("SALES", 
```

Comment:
Should look like below:
    sqlEnv = sqlEnv.registerTable("SALES", new FileEndpoint("/tmp/output", "out.file", 
                                                                                               new CSVMessageFormat(<<csvSchema>>)));